inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/resource/queue/pulsar/PulsarUtils.java

```diff
@@ -39,54 +66,921 @@ public class PulsarUtils {
+        int endIdx = size + buffer.position();
+        while (buffer.position() < endIdx) {
+            int tag = readVarInt(buffer);
+            switch (tag) {
+                case 10:
+                    int _propertiesSize = readVarInt(buffer);
+                    parseFrom(metadata, buffer, _propertiesSize);
+                    break;
+                case 18:
+                    int _partitionKeyBufferLen = readVarInt(buffer);
+                    byte[] partitionKeyArray = new byte[_partitionKeyBufferLen];
+                    buffer.get(partitionKeyArray);
+                    metadata.setPartitionKey(new String(partitionKeyArray));
+                    break;
+                case 24:
+                    int payloadSize = readVarInt(buffer);
+                    metadata.setPayloadSize(payloadSize);
+                    break;
+                case 32:
+                    boolean compactedOut = readVarInt(buffer) == 1;
+                    metadata.setCompactedOut(compactedOut);
+                    break;
+                case 40:
+                    long eventTime = readVarInt64(buffer);
+                    metadata.setEventTime(eventTime);
+                    break;
+                case 48:
+                    boolean partitionKeyB64Encoded = readVarInt(buffer) == 1;
+                    metadata.setPartitionKeyB64Encoded(partitionKeyB64Encoded);
+                    break;
+                case 58:
+                    int _orderingKeyLen = readVarInt(buffer);
+                    byte[] orderingKeyArray = new byte[_orderingKeyLen];
+                    metadata.setOrderingKey(orderingKeyArray);
+                    break;
+                case 64:
+                    long sequenceId = readVarInt64(buffer);
+                    metadata.setSequenceId(sequenceId);
+                    break;
+                case 72:
+                    boolean nullValue = readVarInt(buffer) == 1;
+                    metadata.setNullValue(nullValue);
+                    break;
+                case 80:
+                    boolean nullPartitionKey = readVarInt(buffer) == 1;
+                    metadata.setNullPartitionKey(nullPartitionKey);
+                    break;
+                default:
+                    skipUnknownField(tag, buffer);
+            }
+        }
     }
 
     /**
-     * Get pulsar cluster service url.
+     * Copy from readVarInt method of org.apache.pulsar.common.api.proto.LightProtoCodec class.
+     *
+     * @param buf
+     * @return
      */
-    public static String getServiceUrl(PulsarAdmin pulsarAdmin, String pulsarCluster) throws PulsarAdminException {
-        return pulsarAdmin.clusters().getCluster(pulsarCluster).getServiceUrl();
+    private static int readVarInt(ByteBuffer buf) {
+        byte tmp = buf.get();
+        if (tmp >= 0) {
+            return tmp;
+        } else {
+            int result = tmp & 127;
```

Comment:
Can we avoid using so many if-else here.

Reply:
These codes are copied from readVarInt method of org.apache.pulsar.common.api.proto.LightProtoCodec class，I think it's best to be consistent with the original codes.